index.js

```diff
@@ -72,6 +72,23 @@ module.exports = {
     "simple-import-sort/imports": "error",
     "simple-import-sort/exports": "error",
     "sort-destructure-keys/sort-destructure-keys": "error",
+    /**
+     * List of sonarjs rules we turned off after release of eslint-plugin-sonarjs v2
+     * Some rules are too restrictive for common use cases,
+     * and others are already covered by other plugins used here.
+     */
+    "sonarjs/different-types-comparison": "off",
+    "sonarjs/function-return-type": "off",
+    "sonarjs/no-clear-text-protocols": "warn",
+    "sonarjs/no-ignored-exceptions": "off",
+    "sonarjs/no-misused-promises": "off",
```

Comment:
This rule can be helpful for surfacing incorrect type definitions, e.g. `() => void` vs. `() => Promise<void>`.

It is not, strictly speaking, a type error, but can be helpful in ensuring the definitions we are writing are as accurate as possible (similar to [different-types-comparisons](https://github.com/open-turo/eslint-config-typescript/pull/324/files#r1757299729)).